tensorflow/src/TensorFlow/Output.hs

```diff
@@ -154,3 +155,8 @@ instance IsString Output where
         _ -> Output 0 $ assigned s
         where assigned n = Rendered $ def & name .~ Text.pack n
 
+
+-- | Represents an opaque handle to a mutable resource in the graph.
+-- Typical such resources are variables. The type parameter
+-- corresponds to the dtype of the tensor held in the variable.
+data ResourceHandle dtype = ResourceHandle Output
```

Comment:
newtype?  Also maybe 'a' instead of 'dtype' in the signature to match the other datatypes in this file; up to you.